### PR TITLE
feat: add allowPassphrase to docker config file

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

**Review by jansegre — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

File: config.js.docker [hunk 1]
```diff
@@ -8,6 +8,29 @@ function getParam(name, _default) {
...
+function getBooleanParam(name, _default) {
+  const param = getParam(name, _default);
+  if (param === 'true' || param === '1') {
+    // Should we allow more representations?
+    return true;
+  } else if (param === 'false' || param === '0') {
+    // Should we allow more representations?
+    return false;
+  } else {
+    return _default;
```
Comment (msbrogli) [inline config.js.docker:30]: I'd say throw an exception to prevent running with wrong parameters.

File: config.js.docker [hunk 2]
```diff
@@ -8,6 +8,29 @@ function getParam(name, _default) {
   return argv[name] || process.env[envname] || _default;
 }
 
+function getIntParam(name, _default) {
+  const param = getParam(name, _default);
+  const parsedParam = parseInt(param);
+  if (Number.isInteger(parsedParam)) {
```
Comment (msbrogli) [inline config.js.docker:14]: I feel `parseInt()` will always return an int. Should we prevent things like `"123a"`?

File: config.js.docker [hunk 3]
```diff
@@ -8,6 +8,29 @@ function getParam(name, _default) {
...
+    return parsedParam;
+  } else {
+    return _default;
```
Comment (msbrogli) [inline config.js.docker:17]: I'd say throw an exception to prevent running with wrong parameters.

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: config.js.docker (hunk 1 above)
Comment (pedroferreira1) [reply config.js.docker:30]: If this is an optional parameter we can't accept only 'true', 'false', '0' and '1'. We must accept the default parameter, which is `undefined` or `null`. That's why I kept the `else`

File: config.js.docker (hunk 3 above)
Comment (pedroferreira1) [reply config.js.docker:17]: If this is an optional parameter we must allow the default as `undefined` or `null`. Should we start with this validation in the first line?

```
if (param === undefined || param === null) return param;
```

File: config.js.docker (hunk 2 above)
Comment (pedroferreira1) [reply config.js.docker:14]: Added this commit https://github.com/HathorNetwork/hathor-wallet-headless/pull/82/commits/8e1208b58469e937e96363ab4f25dc1c15b1c3bb. What do you think?

File: config.js.docker (hunk 3 above)
Comment (msbrogli) [reply config.js.docker:22]: Good one. Thanks!

File: config.js.docker (hunk 1 above)
Comment (msbrogli) [reply config.js.docker:35]: But you also return the default value for any random string (e.g. "a").

**Review by msbrogli — APPROVED** [review]

File: config.js.docker (hunk 1 above)
Comment (pedroferreira1) [reply config.js.docker]: Yeah, I've done the same validation for the integer and boolean methods. Can you see this commit? https://github.com/HathorNetwork/hathor-wallet-headless/pull/82/commits/212631446fdf39a2d87ca54e2c499cbc9884b8c0

File: config.js.docker (hunk 3 above)
Comment (pedroferreira1) [reply config.js.docker]: Yeah, I've done the same validation for the integer and boolean methods. Can you see this commit? https://github.com/HathorNetwork/hathor-wallet-headless/pull/82/commits/212631446fdf39a2d87ca54e2c499cbc9884b8c0

**Review by msbrogli — APPROVED** [review]

**Pull request merged**